homeassistant/components/intellifire/config_flow.py

```diff
@@ -127,3 +141,42 @@ async def async_step_user(
             return await self.async_step_pick_device()
         LOGGER.debug("Running Step: manual_device_entry")
         return await self.async_step_manual_device_entry()
+
+    async def async_step_dhcp(self, discovery_info: DhcpServiceInfo) -> FlowResult:
+        """Handle DHCP Discovery."""
+
+        # Run validation logic on ip
+        host = discovery_info.ip
+
+        self._async_abort_entries_match({CONF_HOST: host})
+        try:
+            serial = await validate_host_input(host)
+        except (ConnectionError, ClientConnectionError):
+            return self.async_abort(reason="not_intellifire_device")
+
+        await self.async_set_unique_id(serial)
+        self._abort_if_unique_id_configured(updates={CONF_HOST: host})
+        self._discovered_hosts.append(DiscoveredHostInfo(ip=host, serial=serial))
```

Comment:
I would probably set this instead of appending it to make it clear it's the only one